fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -869,6 +873,20 @@ public CommandProcessingResult undoLoanDisbursal(final Long loanId, final JsonCo
             final Map<String, Object> accountingBridgeData = loan.deriveAccountingBridgeData(currency.getCode(), existingTransactionIds,
                     existingReversedTransactionIds, isAccountTransfer);
             journalEntryWritePlatformService.createJournalEntriesForLoan(accountingBridgeData);
+
+            // Remove All the Disbursement Details If the Loan Product is disabled and exists one
+            if (loan.loanProduct().isDisallowExpectedDisbursements()) {
```

Comment:
If we have multiple disbursement details and the loan is not getting back to approved (by undoing the latest disbursement) i dont think it is a good idea to remove all of the entries.

Reply:
Those now are marked as reversed and not more removed physically